oauth2/src/main/scala/tsec.oauth2/provider/DataHandler.scala

```diff
@@ -0,0 +1,40 @@
+package tsec.oauth2.provider
+
+import java.util.Date
+
+import scala.concurrent.duration._
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Provide accessing to data storage for using OAuth 2.0.
+ */
+trait DataHandler[U] extends AuthorizationHandler[U] with ProtectedResourceHandler[U]
+
+/**
+ * Access token
+ *
+ * @param token Access token is used to authentication.
+ * @param refreshToken Refresh token is used to re-issue access token.
+ * @param scope Inform the client of the scope of the access token issued.
+ * @param life Life of the access token since its creation.
+ * @param createdAt Access token is created date.
+ * @param params Additional parameters to add information/restriction on given Access token.
+ */
+final case class AccessToken(token: String, refreshToken: Option[String], scope: Option[String], life: Option[FiniteDuration], createdAt: Date, params: Map[String, String] = Map.empty[String, String]) {
+  def isExpired: Boolean = expiresIn.exists(_.toMillis < 0)
+
+  val expiresIn: Option[FiniteDuration] = life map { l =>
+    val expTime = createdAt.getTime + l.toMillis
+    ((expTime - System.currentTimeMillis) / 1000) milli
```

Comment:
This thing will break RT _hard_ if this is evaluated wrong.

Anything that checks the system time is an effect. at _least_ `isExpired` should check the time every time, whether you choose to do it in an effect or not.

However, I do believe _isExpired_ should be `F[Boolean]` with `implicit F: Sync[F]` to account for the system check.